src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -279,54 +276,53 @@ private void SetupInvokeThread(Thread invokeThread, bool changeName)
         private FlowControlException InvokeHelper()
         {
             FlowControlException flowControlException = null;
-
             PipelineProcessor pipelineProcessor = null;
+
             try
             {
-#if TRANSACTIONS_SUPPORTED
-                // 2004/11/08-JeffJon
-                // Transactions will not be supported for the Exchange release
-
-                // Add the transaction to this thread
-                System.Transactions.Transaction.Current = this.LocalRunspace.ExecutionContext.CurrentTransaction;
-#endif
                 // Raise the event for Pipeline.Running
                 RaisePipelineStateEvents();
 
                 // Add this pipeline to history
                 RecordPipelineStartTime();
 
-                // Add automatic transcription, but don't transcribe nested commands
-                if (this.AddToHistory || !IsNested)
+                // Add automatic transcription when it's NOT a pulse pipeline, but don't transcribe nested commands.
+                if (!IsPulsePipeline && (AddToHistory || !IsNested))
                 {
-                    bool needToAddOutDefault = true;
-                    CommandInfo outDefaultCommandInfo = new CmdletInfo("Out-Default", typeof(Microsoft.PowerShell.Commands.OutDefaultCommand), null, null, null);
-
-                    foreach (Command command in this.Commands)
+                    foreach (Command command in Commands)
                     {
-                        if (command.IsScript && (!this.IsPulsePipeline))
+                        if (command.IsScript)
                         {
                             // Transcribe scripts, unless they are the pulse pipeline.
-                            this.Runspace.GetExecutionContext.EngineHostInterface.UI.TranscribeCommand(command.CommandText, null);
+                            Runspace.GetExecutionContext.EngineHostInterface.UI.TranscribeCommand(command.CommandText, invocation: null);
                         }
+                    }
+
+                    if (Runspace.GetExecutionContext.EngineHostInterface.UI.IsTranscribing)
+                    {
+                        bool needToAddOutDefault = true;
+                        Command lastCommand = Commands[Commands.Count - 1];
```

Comment:
Actually, thinking about this some more, this assumes that commands are piped together so that only the last one requires 'Out-Default' appended.  But what if the commands include statement separators?  It seems like 'Out-Default' would have to go at the end of those.

Please add a comment that illustrates when 'Out-Default -Transcribe' is needed.

Reply:
I think if the commands include statement separators they'd be considered a completely distinct pipeline, wouldn't they?

Reply:
Right, they would be considered distinct pipelines. Take `a | b; c` as an instance, initially there will be one pipeline processor (1) with one script block command `{ a | b; c }`, while executing that pipeline, another pipeline processor (2) is created with the commands `a` and `b`, after (2) finishes, another pipeline (3) is created for the command `c`.

Reply:
Ok, thanks.  Just wanted to check.